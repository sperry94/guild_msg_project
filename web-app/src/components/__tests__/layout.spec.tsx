import React from "react"
import renderer from "react-test-renderer"
import Layout from "../layout"

describe("Layout", () => {
  it("renders with classes", () => {
    const tree = renderer
      .create(
        <Layout
          classes={{
            main: "my-class",
          }}
        >
          <div>Hello</div>
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders without classes", () => {
    const tree = renderer
      .create(
        <Layout>
          <div>Hello there</div>
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
