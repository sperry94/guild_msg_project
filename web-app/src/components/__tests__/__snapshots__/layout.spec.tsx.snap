// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders with classes 1`] = `
<div
  className="makeStyles-mainInner-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-2 MuiAppBar-colorDefault mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-header-3 MuiToolbar-gutters"
    >
      <nav
        className="makeStyles-nav-4"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <h1
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Guild Chat
          </h1>
        </a>
      </nav>
      <div />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Log In
        </span>
      </button>
    </div>
  </header>
  <main
    className="my-class"
  >
    <div>
      Hello
    </div>
  </main>
</div>
`;

exports[`Layout renders without classes 1`] = `
<div
  className="makeStyles-mainInner-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-2 MuiAppBar-colorDefault mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-header-3 MuiToolbar-gutters"
    >
      <nav
        className="makeStyles-nav-4"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <h1
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Guild Chat
          </h1>
        </a>
      </nav>
      <div />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Log In
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <main>
    <div>
      Hello there
    </div>
  </main>
</div>
`;
